src/main/java/org/stroganov/dao/SoapServiceLibraryDAO.java

```diff
@@ -0,0 +1,195 @@
+package org.stroganov.dao;
+
+import jakarta.xml.ws.WebServiceClient;
+import org.stroganov.entities.*;
+import org.stroganov.wsClient.BookArray;
+import org.stroganov.wsClient.IOException_Exception;
+import org.stroganov.wsClient.LibraryServerServiceImplService;
+import org.stroganov.wsClient.LibraryService;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.util.List;
+
+@WebServiceClient(name = "Client", targetNamespace = "http://example.org")
+public class SoapServiceLibraryDAO implements LibraryDAO {
+
+    private static SoapServiceLibraryDAO instance;
+    private final LibraryService libraryService;
+
+    public SoapServiceLibraryDAO() throws MalformedURLException {
+        // QName qname = new QName("http://ws.stroganov.org/", "LibraryServerServiceImplService");
+        // URL url = new URL("http://127.0.0.1:8082/LibraryService/services?wsdl");
+        // Service service = Service.create(url, qname);
+        // libraryService = service.getPort(LibraryClientService.class);
+
+        LibraryServerServiceImplService libraryServerServiceImplService = new LibraryServerServiceImplService();
+        libraryService = libraryServerServiceImplService.getLibraryServerServiceImplPort();
+    }
+
+    public static synchronized SoapServiceLibraryDAO getInstance() throws MalformedURLException {
+        if (instance == null) {
+            instance = new SoapServiceLibraryDAO();
+        }
+        return instance;
+    }
```

Comment:
This is not quite the correct implementation of the single tone pattern.
Check this url
https://stackoverflow.com/questions/18093735/double-checked-locking-in-singleton
```suggestion
       public static SoapServiceLibraryDAO getInstance() {
        if (instance == null) {
            synchronized (SoapServiceLibraryDAO.class){
                if (instance == null) {
                    instance = new SoapServiceLibraryDAO();
                }
            }
        }
        return instance;
    }
```